funfact/lang/interpreter/_type_deduction.py

```diff
@@ -208,6 +221,17 @@ def ein(
         ╚═════╬══════╝     ║
               ╚════════════╝
         '''
+        # replace ellipsis by anonymous indices
+        num_ell = max(
+            len(lhs.shape) - len(lhs.live_indices),
```

Comment:
I think this would likely have been fine. It would have generated the anonymous indices (it always generates at least 1 as the ellipsis can take the place of just one index), but it would not have inserted them as no ellipsis would be present.